core/src/banking_stage/packet_receiver.rs

```diff
@@ -8,50 +8,85 @@ use {
         unprocessed_transaction_storage::UnprocessedTransactionStorage,
     },
     crossbeam_channel::RecvTimeoutError,
-    solana_measure::measure::Measure,
-    solana_sdk::{
-        saturating_add_assign,
-        timing::{duration_as_ms, timestamp},
-    },
-    std::{
-        sync::atomic::Ordering,
-        time::{Duration, Instant},
-    },
+    solana_measure::{measure, measure::Measure, measure_us},
+    solana_sdk::{saturating_add_assign, timing::timestamp},
+    std::{sync::atomic::Ordering, time::Duration},
 };
 
 pub struct PacketReceiver;
 
 impl PacketReceiver {
-    #[allow(clippy::too_many_arguments)]
     /// Receive incoming packets, push into unprocessed buffer with packet indexes
     pub fn receive_and_buffer_packets(
         packet_deserializer: &mut PacketDeserializer,
-        recv_start: &mut Instant,
-        recv_timeout: Duration,
         id: u32,
         unprocessed_transaction_storage: &mut UnprocessedTransactionStorage,
         banking_stage_stats: &mut BankingStageStats,
         tracer_packet_stats: &mut TracerPacketStats,
         slot_metrics_tracker: &mut LeaderSlotMetricsTracker,
     ) -> Result<(), RecvTimeoutError> {
-        let mut recv_time = Measure::start("receive_and_buffer_packets_recv");
-        let ReceivePacketResults {
+        let (result, recv_time_us) = measure_us!({
+            let recv_timeout = Self::get_receive_timeout(unprocessed_transaction_storage);
+            let mut recv_and_buffer_measure = Measure::start("recv_and_buffer");
+            packet_deserializer
+                .receive_packets(
+                    recv_timeout,
+                    unprocessed_transaction_storage.max_receive_size(),
+                )
+                .map(|receive_packet_results| {
```

Comment:
If we received packets (no error), then we buffer and increment `banking_stage_stats`

Reply:
nit: a bit arbitrary discretion, but i think spelling out return type `-> ()` here for this closure, instead of inference might be helpful for humans.

Reply:
... so that to explicitly indicate this closure is consuming the Ok variant at front.

Reply:
Happy with that indication, but gonna (selfishly) separate PR it. I don't want this to sit for longer and bitrot longer than it already has; now that formal approval is needed to merge.